lib/rules/init-declarations.js

```diff
@@ -105,15 +107,39 @@ module.exports = {
 		const mode = context.options[0] || MODE_ALWAYS;
 		const params = context.options[1] || {};
 
+		// Track whether we're inside a declared namespace
+		let insideDeclaredNamespace = false;
+
 		//--------------------------------------------------------------------------
 		// Public API
 		//--------------------------------------------------------------------------
 
 		return {
+			TSModuleDeclaration(node) {
+				if (node.declare) {
+					insideDeclaredNamespace = true;
+				}
+			},
+
+			"TSModuleDeclaration:exit"(node) {
+				if (node.declare) {
+					insideDeclaredNamespace = false;
```

Comment:
[Testing] Removing this line doesn't fail any tests right now. But it's definitely an important line to have for cases of variables after a declared namespace:

```ts
declare namespace myLib {
    let valueInside: number;
}
let valueOutside: number;
```

Reply:
Yeah, added more test cases in https://github.com/eslint/eslint/pull/19540/commits/9918115cc5dc0f0616f7677b371d6b993eb43945